src/cgtop/cgtop.c

```diff
@@ -864,6 +871,13 @@ static int get_cgroup_root(char **ret) {
         const char *m;
         int r;
 
+        if (arg_root) {
+                *ret = strdup(arg_root);
+                if (!*ret)
```

Comment:
humpf, we generally avoid clobbering call-by-ref parameters on failure. See CODING_STYLE. Hence, please don't override `*ret` until your know you succeeded. Specifically, write the result into a temporary variable and only after the oom check succeeded write it into *ret...